lego/apps/events/constants.py

```diff
@@ -83,7 +83,7 @@
 
 PAYMENT_PENDING = "pending"
 PAYMENT_SUCCESS = "succeeded"
-PAYMENT_FAILURE = "failed"  # Matches stripe API
+PAYMENT_FAILURE = "payment_failed"  # Matches stripe API
```

Comment:
Do we have to migrate old payments if they are stored as failed payments?

Reply:
I realized this is actually wrong: https://stripe.com/docs/payments/intents#intent-statuses
I'm not sure how to handle this, since the payment intent does not have a `failed` status. We could set the status to failed if the charge object has a `failed` status, since we dont use the same payment intent to retry payments. Not the best solution though.
Or we could try to detect when a payment intent has been tried once, and manually cancel it?
Thoughts?

Reply:
Hmm.

I haven't look that much at Payment intents, but in the link :arrow_up: it looks like the flow is quite different from the current one. If that _is_ the case, I think we should rethink the payment process instead of just fitting it into the previous model.

Also, just like @JonasBak says, we would have to migrate the data no matter what. We also need to fix https://github.com/webkom/lego/issues/1554 somehow. It will be bad S if we suddenly send out thousands of mails saying that people have to pay for stuff they have already paid for.

Reply:
The flow actually isn't __that__ much different. The difference is that intents allow for a lot more flexibility. For our payment flow, however, using basically the same flow as before allows for minimal changes and keeps the process simple. The difference in the current implementation is that the Charge is contained in the payment intent.

The states pictured in the link is all handled by Stripe automatically, we shouldn't need to take action in any other case than if the intent failed or succeeded. i.e for us the flow is as simple as:
1. creating a payment intent on the server
2. attaching a payment method on the client
4. We monitor the state with webhooks,  and check whether the payment failed or succeeded.

If we use the `payment_failed` [webhook event](https://stripe.com/docs/payments/payment-intents/verifying-status#handling-webhook-events) to determine if the payment failed, we should have pretty much the same flow, and it should remove the need for migrating since we still only use the single `charge`, and as far as I see, there is no reason for the `PAYMENT_FAILURE` to match Stripe?
This, of course, does not fix the issue with the old payments, and may not be the best solution, but it does work just fine.

Reply:
> The flow actually isn't that much different.

It is still different. A lot of the phrasing doesn't make sense with the new API (ref. comments from @orhanhenrik). I guess we can draw a simple chart on Wednesday, so we all agree on the flow, and on how it should be implemented.

> there is no reason for the PAYMENT_FAILURE to match Stripe?

Neh, in theory no, as long as we make sure that we always convert the values.

> This, of course, does not fix the issue with the old payments, and may not be the best solution, but it does work just fine.

Ofc, but it is related to the code this PR changes :man_shrugging: